app/controllers/nonsipp/membertransferout/TransfersOutCYAController.scala

```diff
@@ -123,32 +163,72 @@ object TransfersOutCYAController {
+    optYear: Option[String] = None,
+    optCurrentVersion: Option[Int] = None,
+    optPreviousVersion: Option[Int] = None,
+    compilationOrSubmissionDate: Option[LocalDateTime] = None
   ): FormPageViewModel[CheckYourAnswersViewModel] = {
-    val heading: InlineMessage = mode.fold(
-      normal = "checkYourAnswers.heading",
-      check = Message("transfersOutCYAController.heading.check", memberName)
-    )
+    val heading: InlineMessage =
+      mode
+        .fold(
+          normal = "checkYourAnswers.heading",
+          check = Message("transfersOutCYAController.heading.check", memberName),
+          viewOnly = "transfersOutCYAController.viewOnly.title"
+        )
 
     FormPageViewModel[CheckYourAnswersViewModel](
-      title = "checkYourAnswers.title",
+      mode = mode,
+      title = mode
+        .fold(
+          normal = "checkYourAnswers.title",
+          "",
+          viewOnly = "checkYourAnswers.viewOnly.title"
+        ),
       heading = heading,
       description = Some(ParagraphMessage("transfersOutCYA.paragraph")),
       page = CheckYourAnswersViewModel(
-        sections = rows(srn, memberName, index, journeys),
+        sections = rows(srn, memberName, index, journeys, mode match {
+          case ViewOnlyMode => NormalMode
+          case _ => mode
+        }),
         inset = Option.when(journeys.size == 5)("transfersOutCYAController.inset")
       ),
       refresh = None,
-      buttonText = "site.continue",
-      onSubmit = routes.TransfersOutCYAController.onSubmit(srn, index, mode)
+      buttonText = mode.fold(normal = "site.continue", check = "site.continue", viewOnly = "site.return.to.tasklist"),
+      onSubmit = controllers.nonsipp.membertransferout.routes.TransfersOutCYAController.onSubmit(srn, index, mode),
+      optViewOnlyDetails = if (mode == ViewOnlyMode) {
+        Some(
+          ViewOnlyDetailsViewModel(
+            updated = viewOnlyUpdated,
+            link = None,
+            submittedText = Some(Message("")),
+            title = "transfersOutCYAController.viewOnly.title",
+            heading = Message("transfersOutCYAController.viewOnly.heading", memberName),
+            buttonText = "site.return.to.tasklist",
+            onSubmit = (optYear, optCurrentVersion, optPreviousVersion) match {
+              case (Some(year), Some(currentVersion), Some(previousVersion)) =>
+                controllers.nonsipp.membertransferout.routes.TransfersOutCYAController
+                  .onSubmitViewOnly(srn, year, currentVersion, previousVersion)
+              case _ =>
+                controllers.nonsipp.membertransferout.routes.TransfersOutCYAController
+                  .onSubmit(srn, index, mode)
+            }
+          )
+        )
+      } else {
+        None
+      }
     )
   }
 
   private def rows(
     srn: Srn,
     memberName: String,
     index: Max300,
-    journeys: List[TransfersOutCYA]
+    journeys: List[TransfersOutCYA],
+    mode: Mode
```

Comment:
Looks unused

Reply:
The above mode I added because of I added in 'row', 

        sections = rows(srn, memberName, index, journeys, `mode `match {
          case ViewOnlyMode => NormalMode
          case _ => mode
        }),

Please correct me, if it is wrong.

Reply:
Is `mode` used in `rows` method?

Reply:
Yes, Kornel